packages/alert/src/Alert.js

```diff
@@ -3,40 +3,63 @@ import * as React from "react";
 import PropTypes from "prop-types";
 import Button, { buttonTypes, buttonSizes } from "@crave/farmblocks-button";
 
-import Container from "./styledComponents/Container";
-import DismissButtonContainer from "./styledComponents/DismissButtonContainer";
+import StyledAlert from "./styledComponents/Alert";
 import AlertTypes from "./constants/alertTypes";
+import BrieflyDisplay from "./BrieflyDisplay";
 
-// @FIXME annotating props as Object while we dont have a way to prevent redundant type anotation.
-// see issue #22
-const Alert = (props: Object) => (
-  <Container type={props.type}>
-    <p>{props.text}</p>
+class Alert extends React.Component<Object, Object> {
+  static propTypes = {
+    text: PropTypes.string.isRequired,
+    type: PropTypes.oneOf(Object.keys(AlertTypes)),
+    dismissable: PropTypes.bool,
+    onDismiss: PropTypes.func,
+    visibleTime: PropTypes.number,
+    zIndex: PropTypes.number
+  };
 
-    {props.dismissable && (
-      <DismissButtonContainer>
-        <Button
-          icon="wg-close"
-          size={buttonSizes.SMALL}
-          type={buttonTypes.OFF_NEUTRAL}
-          onClick={props.onDismiss}
-        />
-      </DismissButtonContainer>
-    )}
-  </Container>
-);
+  static defaultProps = {
+    type: AlertTypes.NEWS,
+    dismissable: true,
+    onDismiss: () => null
+  };
 
-Alert.propTypes = {
-  text: PropTypes.string.isRequired,
-  type: PropTypes.oneOf(Object.keys(AlertTypes)),
-  dismissable: PropTypes.bool,
-  onDismiss: PropTypes.func
-};
+  constructor(props: Object) {
+    super(props);
+    this.state = {
+      isVisible: true
+    };
+  }
 
-Alert.defaultProps = {
-  type: AlertTypes.NEWS,
-  dismissable: true,
-  onDismiss: () => null
-};
+  render() {
+    if (!this.state.isVisible) {
+      return null;
+    }
+    const { visibleTime, zIndex } = this.props;
+    const dismissHandler = () => {
+      this.setState({ isVisible: false });
+      this.props.onDismiss();
+    };
+    const alert = (
+      <StyledAlert type={this.props.type} {...(zIndex ? { zIndex } : {})}>
```

Comment:
Hmmm can we pass zIndex like this?

`<StyledAlert type={this.props.type} zIndex={this.props.zIndex}>`

Reply:
it may require a small change on the styled component, but sure